streams/src/main/java/org/apache/kafka/streams/processor/internals/ProcessorStateManager.java

```diff
@@ -129,50 +129,52 @@ public void register(StateStore store, boolean loggingEnabled, StateRestoreCallb
         }
         
         // check that the underlying change log topic exist or not
-        String topic;
+        String topic = null;
         if (loggingEnabled) {
             topic = storeChangelogTopic(this.applicationId, store.name());
         } else if (sourceStoreToSourceTopic != null && sourceStoreToSourceTopic.containsKey(store.name())) {
             topic = sourceStoreToSourceTopic.get(store.name());
-        } else {
-            throw new IllegalArgumentException("Store is neither built from source topic, nor has a changelog.");
         }
 
-        // block until the partition is ready for this state changelog topic or time has elapsed
-        int partition = getPartition(topic);
-        boolean partitionNotFound = true;
-        long startTime = System.currentTimeMillis();
-        long waitTime = 5000L;      // hard-code the value since we should not block after KIP-4
-
-        do {
-            try {
-                Thread.sleep(50L);
-            } catch (InterruptedException e) {
-                // ignore
-            }
+        if (topic != null) {
```

Comment:
I think i said this before, i think it is cleaner to bail out early if the topic is null. We then don't need to check again and for the main case, i.e, topic is not null, the code is outside of an `if` statement
```
if (topic == null) {
  this.stores.put(storeName, store);
  return;
}
```